drivers/rtc/rtc_rv3028.c

```diff
@@ -260,18 +203,22 @@ static int rv3028_eeprom_wait_busy(const struct device *dev)
 			return -ETIME;
 		}
 
-		k_busy_wait(RV3028_EEBUSY_POLL_US);
+		if (k_is_pre_kernel()) {
+			k_busy_wait(poll_ms * 1000);
+		} else {
+			k_msleep(poll_ms);
+		}
```

Comment:
Updated `rv3028_eeprom_wait_busy` to use `k_msleep` if the kernel is initialized.